Messaging.md

```diff
@@ -44,15 +55,102 @@ The following convenience types are also defined:
+
+### The contract_info Message
+
+This message contains information about a contracts outcomes and their corresponding payouts. To save space, only one side's POV is included in this message as the other can be derived using `remote_payout = total_collateral - local_payout`.
+
+#### Version 0 `contract_info`
+
+1. type: 42768 (`contract_info_v0`)
+2. data:
+   * [`sha256`:`outcome_1`]
+   * [`u64`:`outcome_1_local_payout`]
+   * ...
+   * [`sha256`:`outcome_n`]
+   * [`u64`:`outcome_n_local_payout`]
+
+This type of contract info is a simple enumeration where the value `n` is omitted from being explicitly included as it can be derived from the length field of the TLV.
+
+### The `oracle_info` Message
+
+This message contains information about the oracle(s) to be used in executing a DLC, and possibly the outcomes possible if these are not specified in the corresponding `contract_info`.
+
+#### Version 0 `oracle_info`
+
+1. type: 42770 (`oracle_info_v0`)
+2. data:
+   * [`x_point`:`oracle_public_key`]
+   * [`x_point`:`oracle_nonce`]
+
+This type of oracle info is for single-oracle, single signature (and hence single nonce) events.
+
+### The `funding_input` Message
+
+This message contains information about a specific input to be used in a funding transaction, as well as its corresponding on-chain UTXO.
+
+#### Temporary `funding_input`
+
+1. type: 42772 (`funding_input_temp`)
+2. data:
+   * [`sha256`:`prev_txid`]
+   * [`u32`:`vout`]
+   * [`u64`:`value`]
+   * [`spk`:`spk`]
+
+This type is not actually being proposed and will be replaced with a more suitable type in the near future. This is however needed to get a start on static test vectors and compatibility between implementations so it is currently included here for use in test vectors (and will later be replaced both here and in test vectors).
+
+`prev_txid` is the little-endian TXID of the transaction on which this UTXO resides at `vout`.
+`value` is the output amount of the UTXO and `spk` is the script public key that must be satisfied by this input.
+
+Note that `prev_txid || vout` form a serialized Transaction Outpoint and ` value || spk` form a serialized Transaction Output
+so that the data on this TLV can be parsed by most libraries conveniently as `outpoint || output`.
+
+### The `cet_signatures` Message
+
+This message contains CET signatures and any necessary information linking the signatures to their corresponding outcome.
+
+#### Version 0 `cet_signatures`
+
+1. type: 42774 (`cet_signatures_v0`)
+2. data:
+   * [`adaptor_signature`:`signature_1`]
+   * ...
+   * [`adaptor_signature`:`signature_n`]
+
+This type should be used with [`contract_info_v0`](#version-0-contract_info) where each indexed signature in the data corresponds to the outcome of the same index. And as in [`contract_info_v0`](#version-0-contract_info), the number of signatures is omitted as it can be derived from the length field of the TLV.
```

Comment:
>. And as in contract_info_v0

Shouldn't start a sentence with `And` maybe change to `Following contract_info_v0, the ...`